src/app/clusters/valve-configuration-and-control-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "CodegenIntegration.h"
+#include <app/clusters/time-synchronization-server/CodegenIntegration.h>
+#include <app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-cluster.h>
+#include <app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-delegate.h>
+#include <app/static-cluster-config/ValveConfigurationAndControl.h>
+#include <app/util/attribute-storage.h>
+#include <data-model-providers/codegen/ClusterIntegration.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ValveConfigurationAndControl;
+using namespace chip::app::Clusters::ValveConfigurationAndControl::Attributes;
+
+namespace {
+
+constexpr size_t kValveConfigurationAndControlFixedClusterCount =
+    ValveConfigurationAndControl::StaticApplicationConfig::kFixedClusterConfig.size();
+constexpr size_t kValveConfigurationAndControlMaxClusterCount =
+    kValveConfigurationAndControlFixedClusterCount + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+LazyRegisteredServerCluster<ValveConfigurationAndControlCluster> gServers[kValveConfigurationAndControlMaxClusterCount];
+
+class CodegenTimeSyncTracker : public TimeSyncTracker
+{
+public:
+    bool IsTimeSyncClusterSupported() override { return (nullptr != TimeSynchronization::GetClusterInstance()); }
```

Comment:
If time sync is not compiled in, I would expect the C++ for TimeSync::GetClusterInstance to not be compiled in, so this should give a linker error.

Since we are in ember world, maybe we should use the ember defines (as ugly as this may seem) and constexpr it:

https://github.com/project-chip/connectedhomeip/blob/318d81e0882d02e23f944bd16b6d080fed629368/scripts/tools/zap/tests/outputs/all-clusters-app/app-templates/gen_config.h#L49

it says that `MATTER_DM_TIME_SYNCHRONIZATION_CLUSTER_SERVER_ENDPOINT_COUNT` is a defined constant.